libs/wal_decoder/src/serialized_batch.rs

```diff
@@ -541,6 +541,25 @@ impl SerializedValueBatch {
         })
     }
 
+    /// Generates batch statistics for metrics.
+    pub fn stats(&self) -> SerializedValueStats {
+        let mut stats = SerializedValueStats::default();
+        for metadata in &self.metadata {
+            let ValueMeta::Serialized(valuemeta) = metadata else {
+                continue;
+            };
+            let is_block_key = Key::from_compact(valuemeta.key).is_block_key();
+            let will_init = valuemeta.will_init;
+            match (is_block_key, will_init) {
+                (true, true) => stats.page_images += 1,
+                (true, false) => stats.page_deltas += 1,
+                (false, true) => stats.metadata_images += 1,
+                (false, false) => stats.metadata_deltas += 1,
```

Comment:
Do we use images/deltas for metadata values too, or only for pages?

Also, are we even interested in tracking metadata values? I tried a local `pgbench` run, and there were no metadata values at all.

Reply:
I think metadata is all Value::Image -- certainly things like relation/db existence and relation size are all images.

Reply:
>I tried a local pgbench run, and there were no metadata values at all.

Ok, so I'm doing to counting too far up the stack -- metadata is typically received as `MetadataRecord`, and then translated to metadata keys during ingestions. I'll need to account for these in the `DatadirModification` instead.